Modules/Test/classes/class.ilTestEvaluationGUI.php

```diff
@@ -935,37 +907,24 @@ public function outParticipantsPassDetails()
             $template->parseCurrentBlock();
         }
 
-        $list_of_answers = $this->getPassListOfAnswers($result_array, $active_id, $pass, ilSession::get('tst_results_show_best_solutions'), false, false, false, true, $objectivesList, $testResultHeaderLabelBuilder);
-        $template->setVariable("LIST_OF_ANSWERS", $list_of_answers);
-        $template->setVariable("PASS_DETAILS", $this->ctrl->getHTML($overviewTableGUI));
-
-        $data = $this->object->getCompleteEvaluationData();
-        $result = $data->getParticipant($active_id)->getReached() . " " . strtolower($this->lng->txt("of")) . " " . $data->getParticipant($active_id)->getMaxpoints() . " (" . sprintf("%2.2f", $data->getParticipant($active_id)->getReachedPointsInPercent()) . " %" . ")";
-        $template->setCurrentBlock('total_score');
-        $template->setVariable("TOTAL_RESULT_TEXT", $this->lng->txt('tst_stat_result_resultspoints'));
-        $template->setVariable("TOTAL_RESULT", $result);
-        $template->parseCurrentBlock();
-
-        if (!$this->getObjectiveOrientedContainer()->isObjectiveOrientedPresentationRequired()) {
-            $template->setVariable("USER_DATA", $user_data);
-
-            $uname = $this->object->userLookupFullName($user_id);
-            $template->setVariable("TEXT_HEADING", sprintf($this->lng->txt("tst_result_user_name_pass"), $pass + 1, $uname));
-
-            $template->setVariable("TEXT_RESULTS", $testResultHeaderLabelBuilder->getPassDetailsHeaderLabel($pass + 1));
-        }
+        $title = sprintf(
+            $this->lng->txt("tst_result_user_name_pass"),
+            $pass + 1,
+            ilObjUser::_lookupFullname($this->object->_getUserIdFromActiveId($active_id))
+        );
 
-        $template->setVariable("FORMACTION", $this->ctrl->getFormAction($this));
+        $dic = ilTestDIC::dic();
```

Comment:
Hi @nhaagen 

I looked at this today for another reason (@dsstrassner will write you a feedback concerning this) and I stumbled over this pattern, ...Could we get rid of it? I think this is the very definition of a broken Service-Locator and we are hiding the dependency to 'factory.results' in the far depth of the code. In my ideal world the `ilTestDIC` is only used in  `ilObjTest` and there it also requires no statics to get to it. I know, this might not be possible right now, but we should at least avoid peppering the code with those. So: Could we get rid of the `static` function and try to inject the parts we need?
The longer I think about it the more I come to the conclusion that my definition of a `DIC` is not a "Dependency Injection Container", but a "Dependency Initialization Container".

I'm sure there are interesting discussions to be had here, so I'm looking forward to having them.

Thanks for this important contribution. I will give you a full review once @dsstrassner has accepted the user facing part. Just wanted to start the discussion here right now.

Best,
@kergomard